src/sentry/tasks/relay.py

```diff
@@ -99,79 +97,146 @@ def update_config_cache(
         )
 
 
-def schedule_update_config_cache(
-    generate, project_id=None, organization_id=None, public_key=None, update_reason=None
-):
-    """
-    Schedule the `update_config_cache` with debouncing applied.
+@instrumented_task(
+    name="sentry.tasks.relay.build_config_cache",
+    queue="relay_config",
+    acks_late=True,
+    soft_time_limit=30,
+    time_limit=32,
+)
+def build_config_cache(public_key=None, trigger=None, **kwargs):
+    """Build a project config and put it in the Redis cache.
 
-    See documentation of `update_config_cache` for documentation of parameters.
+    This task is the primary way a new project config is computed.
+
+    :param organization_id: The organization for which to invalidate configs.
+    :param project_id: The project for which to invalidate configs.
+    :param generate: If `True`, caches will be eagerly regenerated, not only
+        invalidated.
+
+    Do not invoke this task directly, instead use :func:`schedule_build_config_cache`.
     """
+    validate_args(public_key=public_key)
 
-    if (
-        settings.SENTRY_RELAY_PROJECTCONFIG_CACHE
-        == "sentry.relay.projectconfig_cache.base.ProjectConfigCache"
-    ):
-        # This cache backend is a noop, don't bother creating a noop celery
-        # task.
-        metrics.incr(
-            "relay.projectconfig_cache.skipped",
-            tags={"reason": "noop_backend", "update_reason": update_reason},
-        )
-        return
+    sentry_sdk.set_tag("update_reason", trigger)
```

Comment:
having these two outside of the `try...finally` is potentially dangerous, if someone schedules a task with the wrong args it would fail and we don't delete the debounce keys.